### ISSUE TITLE
Questtion: getting type argument of parent interface/class

### ISSUE THREAD

**Issue by s0nicyouth** [google/ksp#1077]
Hello,
suppose I have 
```
data class A(
    val a: List<Int>
)
```
List implements Iterable<E> as Iterable<Int>. So how do I get type argument of Iterable specificaly?
I can get argument of List via something like
`a.type.resolve().arguments` but I haven'tfound a way to dig down to Iterable and get it's tpe arguments.

**Comment by neetopia** [comment]
you should be able to get the super types of `List<Int>`, and from there you can find the `Iterable` super type with correct argument.

**Comment by s0nicyouth** [comment]
> you should be able to get the super types of `List<Int>`, and from there you can find the `Iterable` super type with correct argument.

Well, that's what I expected I can do but I haven't found a way yet. Do you know the recipe? I can indeed find super types of List<Int> declarations but I can only find type parameters not type arguments from there.

**Comment by Squiry** [comment]
Plus for that. Not having access to supertypes in `KSType` is kind of annoying. We need kind of `Types.asMemberOf` for type parameters and every element really, resolving "what <T> from `Class<T>` is in `OtherClass : Class<String>` is a nightmare.

**Comment by neetopia** [comment]
sounds like an easier way to do type substitution is wanted here.

**Comment by Squiry** [comment]
Yep, right now I compare two `KSTypeParameter.name` on reference equal to match them, because KSTypeParameter implementations aren't even equals for the same parameter.
